packages/NodeNameResolver/NodeNameResolver.php

```diff
@@ -169,13 +168,24 @@ public function getNames(array $nodes): array
     }
 
     /**
-     * Ends with ucname
-     * Starts with adjective, e.g. (Post $firstPost, Post $secondPost)
+     * @param string|string[] $suffix
      */
-    public function endsWith(string $currentName, string $expectedName): bool
+    public function endsWith(Node $node, string|array $suffix): bool
```

Comment:
Do we need this method at all, or should just use the `matchesName(.., 'myname*')` instead!

Reply:
ohh I see. you want todo it the other way arround. drop all regex magic, in case thats good enough for all existing use-cases

Reply:
You're right. Removing `endsWith()` and `startsWith()` might make more sense. There are only 5 cases in our whole code base, so I think it would be better choice.

Reply:
`startsWith()` seems already used in rector-phpunit and rector-symfony so that should be changed first 

https://github.com/rectorphp/rector-phpunit/commit/f07d7f02e44488a9b022c561623686c460e62002

https://github.com/rectorphp/rector-symfony/commit/b9303b57a4076721d9f83f1ab75e3949c66e7fad